shared/test-helpers/lib/VersionedContractsProvider.js

```diff
@@ -0,0 +1,83 @@
+const fs = require('fs')
+const path = require('path')
+const { spawnSync } = require('child_process')
+
+const APPS_GIT_URL = 'https://github.com/aragon/aragon-apps.git'
+
+module.exports = (web3, TruffleContract, TruffleConfig) => {
+  return class VersionedContractsProvider {
+    static fromBranch({ commit, source = APPS_GIT_URL, projectDir = '.', outputDir = 'tmp' }) {
```

Comment:
Wonder if we could be more generic than using "branch" terminology. I foresee us using tags most of the time, but I imagine there's a better abstraction for git references in general.

Reply:
I started it with a tag/branch abstraction, but I thought that maybe commit was more flexible. We can extend it to whatever we want actually :)